src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,138 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private RegistrationServiceImpl registrationServiceImpl;
+    private User user;
+
+    @BeforeEach
+    void initializeObjects() {
+        registrationServiceImpl = new RegistrationServiceImpl();
+        user = new User();
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_ageLessThan18_notOk() {
+        user.setAge(16);
+        user.setLogin("validLogin");
+        user.setPassword("validPass");
+        assertThrows(InvalidUserDataException.class, () -> registrationServiceImpl.register(user));
+    }
+
+    @Test
+    void register_validUser_ok() {
+        user.setAge(19);
+        user.setLogin("validLogin");
+        user.setPassword("validPass");
+        User registeredUser = registrationServiceImpl.register(user);
+        assertEquals(user, registeredUser);
+    }
+
+    @Test
+    void register_ageEquals18_ok() {
+        user.setAge(18);
+        user.setLogin("validLogin");
+        user.setPassword("validPass");
+        User registeredUser = registrationServiceImpl.register(user);
+        assertEquals(user, registeredUser);
+    }
+
+    @Test
+    void register_LoginExist_notOk() {
+        User existingUser = new User();
+        existingUser.setLogin("validLogin");
+        Storage.people.add(existingUser);
+        user.setAge(20);
+        user.setLogin("validLogin");
+        user.setPassword("validPass");
+        assertThrows(InvalidUserDataException.class, () -> registrationServiceImpl.register(user));
```

Comment:
This test checks if the login already exists in the storage. However, the checklist advises using low-level methods like `Storage.people.add()` directly instead of `RegistrationService.register()` to add users for testing. Please make sure you're not using the `register` method to add test users.